### PR TITLE
CFX fails & timeouts

### PR DESCRIPTION
**Pull request by brohamgoham**
fixing some non deterministic failures, i believe  there is just one issue causing these timeouts
cache-events no longer times out on CFX but does on ETH

### CODE REVIEW

File: js/stdlib/ts/cfxers.ts [hunk 1]
```diff
@@ -181,16 +177,13 @@ export class Contract implements IContract {
...
-            txn.gas = est.gasLimit;
+          if ( txn.gasLimit === undefined ) {
+            txn.gasLimit = est.gasLimit;
           }
           if ( txn.storageLimit === undefined ) {
             txn.storageLimit = est.storageCollateralized;
           }
         }
-        if ( txn.storageLimit === undefined || txn.storageLimit == 0 ) {
-          txn.storageLimit = 2048;
```
Comment (jeapostrophe) [inline js/stdlib/ts/cfxers.ts:193]: In my experience, the calculation of the storage limit estimate does not always put something in when it is needed and if it isn't needed, will get refunded, so we need this to use linear state.

File: js/stdlib/ts/cfxers.ts [hunk 2]
```diff
@@ -181,16 +177,13 @@ export class Contract implements IContract {
         }
         debug(`cfxers:handler`, fname, {est, est_err});
         if ( est ) {
-          if ( txn.gas === undefined ) {
-            txn.gas = est.gasLimit;
+          if ( txn.gasLimit === undefined ) {
+            txn.gasLimit = est.gasLimit;
```
Comment (jeapostrophe) [inline js/stdlib/ts/cfxers.ts:181]: This --- https://confluxnetwork.gitbook.io/js-conflux-sdk/api/transaction#Transaction.js/Transaction/**constructor** --- says the field is named `gas`, not `gasLimit`

**Comment by brohamgoham** [comment]
raffle was running fine on cfx locally but still failing on CI
And mint basic only timed out because ran out of gas

**Comment by jeapostrophe** [comment]
> raffle was running fine on cfx locally but still failing on CI

In my experience, this can mean that there's a race condition where the CI machine running slower than our fast developer laptops behaves differently.

> And mint basic only timed out because ran out of gas

Does it work if the participants have more CFX in their accounts? If the gasLimit is increased? Another thing that can happen is that, at least on ETH, "out of gas" is kind of a catch all that just means "the transaction failed" which could mean a bigger problem (like trying to compete in a finished race)

File: examples/raffle/index.rsh [hunk 3]
```diff
@@ -122,8 +122,9 @@ export const main =
         const sponsort = declassify(_sponsort); });
       Sponsor.publish(sponsort);
       require(sponsortc == digest(sponsort));
-
-      const winningNo = (hwinner + (sponsort % howManyReturned)) % howManyReturned;
+      // CFX does like doing division with UInts );
+      // shared_impl.ts ln 356, Players bigNumber division is throwing it off
+      const winningNo = hwinner + sponsort;
```
Comment (jeapostrophe) [inline examples/raffle/index.rsh:127]: The change that you made is not correct. What you've changed it to will overflow

Comment (brohamgoham) [reply examples/raffle/index.rsh:127]: The assertion of the original mod division remainder (sponsors % howManyReturned) % howmanyReturn 
 fails and points to a BigNumbers mod error, I kept thinking it was a issue w gas and storage but its how CFX handles bigNumber values im pretty sure

Comment (jeapostrophe) [reply examples/raffle/index.rsh:127]: That is very valuable to learn. We can't change the program though. Potentially we need to update how the Reach compiler outputs OR we need to get a problem report to Conflux. IDEA: Could you write a little Solidity program by hand that has the same problem when run on Conflux, but not on Eth?

File: examples/raffle/index.rsh [hunk 4]
```diff
@@ -152,4 +152,4 @@ export const main =
         }))
       .timeout(deadline, () => closeTo(Sponsor, () => {}));
       commit();
-    });
+    });
```
Comment (DanBurton) [inline examples/raffle/index.rsh:155]: Nitpick: can you set your editor to keep a newline at the end of the file?

Comment (brohamgoham) [reply examples/raffle/index.rsh:155]: i got you

File: js/stdlib/ts/cfxers.ts [hunk 5]
```diff
@@ -423,13 +422,31 @@ export class Wallet implements IWallet {
...
+    let newStorageCost = format.big(estimate.storageCollateralized).times(storageRatio).toFixed(0);
+    
+    // Note: balace needs to cover value +  (gas + storageLimit)
+    let gasXstorage = format.big(newGasCost).plus(newStorageCost).toFixed(0);
+    let finalCost = format.big(txn.value).plus(gasXstorage).toFixed(0);
+    const final = BigInt(finalCost);
+    
+    debug(`SendTxn attempt, Final Cost of Tx is , ${final},  Balance of sender ${from} is ${balance}`);
+    if ( final > balance ) {
+      debug(`Checking: Account balanace of  ${from} is ${balance} and gasFee is, ${newGasCost}: Total TxValue is ${final}`)
```
Comment (jeapostrophe) [inline js/stdlib/ts/cfxers.ts:441]: Typo

File: js/stdlib/ts/cfxers.ts [hunk 6]
```diff
@@ -423,13 +422,31 @@ export class Wallet implements IWallet {
...
+      throw Error(` INSUFFICIENT FUNDS GAS COST IS ${newGasCost},  TXN VALUE IS  ${final}, ACCOUNT ${from} ONLY HAS A BALANCE OF ${balance}`);
```
Comment (jeapostrophe) [inline js/stdlib/ts/cfxers.ts:442]: Still no test for this

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

**Pull request merged**